src/pages/blog/coming-soon-threading.md

```diff
@@ -0,0 +1,119 @@
+---
+templateKey: blog-post
+title: Coming Soon - Threaded Ironic
+author: Julia Kreger
+date: 2025-07-25T20:00:00.000Z
+category:
+  - label: Development Update
+    id: category-A7fnZYrE9
+---
+
+To understand where you're going, you need to understand where you started.
+
+We typically don't talk about [eventlet](https://eventlet.readthedocs.io/en/latest/). It was a foundational technology which OpenStack based on to address concurrency in a Python process. It enabled a <!--more--> level of concurrency which has simplified many things for OpenStack without developers needing to be mindful of threading. In essence, Eventlet replaces native threading and numerous blocking calls in a Python application. It also came with caveats, but it provided a suitable substrate to build upon.
+
+For example, calling a time.sleep() call wouldn't actually block the application. The same is true for socket IO, and numerous other code paths which block a "thread".
+
+This is because eventlet quite literally overwrote Threading logic with code which functionally turns the threads into [coroutines](https://en.wikipedia.org/wiki/Coroutine), where any blocking action would result in different coroutines executing.
+
+For OpenStack as a whole, this was fairly useful. Most activities were not bound by CPU resource constraints, even when a process was functionally single threaded for specific actions. Of course, some services decomposed furhter along logical boundries, but overall OpenStack services had found an balance in the process and task execution models which met requirements while also kept the design and maintenance relatively simple.
+
+But time has finally come for OpenStack projects to remove eventlet. The reasons are complex, and truthfully eventlet has been holding OpenStack projects back, while changes in Python make the maintenance of eventlet problematic as time moves forward.
+
+The end result, Ironic is becoming a true multi-threaded application. While not quite already released, expect Ironic 32.0 to be the result of the hard work of contributors in the community.
```

Comment:
There is probably value in at least saying (part of OpenStack 2025.2 release) or similar.